molgenis-data/src/main/java/org/molgenis/data/meta/MetaValidationUtils.java

```diff
@@ -0,0 +1,67 @@
+package org.molgenis.data.meta;
+
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.MolgenisDataException;
+import org.molgenis.fieldtypes.CompoundField;
+
+public class MetaValidationUtils
+{
+	public static final int MAX_ATTRIBUTE_LENGTH = 30;
+
+	/**
+	 * Validates names of entities, packages and attributes. Rules: only [a-zA-Z0-9_] are allowed, name must start with
+	 * a letter
+	 */
+	public static void validateName(String name)
+	{
+		if (name.length() > MAX_ATTRIBUTE_LENGTH)
+		{
+			throw new MolgenisDataException("Attribute name [" + name + "] is too long: maximum length is "
+					+ MAX_ATTRIBUTE_LENGTH + " characters.");
+		}
+
+		if (!name.matches("[a-zA-Z0-9_#]+"))
+		{
+			throw new MolgenisDataException("Invalid characters in: [" + name
+					+ "] Only letters (a-z, A-Z), digits (0-9) and underscores (_) are allowed.");
+		}
+
+		if (Character.isDigit(name.charAt(0)))
+		{
+			throw new MolgenisDataException("Invalid name: [" + name + "] Names must start with a letter.");
```

Comment:
Underscores anywhere will clash with the underscore as a package/entity name delimiter, I think?

> On 05 Jun 2015, at 10:53, tommydeboer <notifications@github.com> wrote:
> 
> In molgenis-data/src/main/java/org/molgenis/data/meta/MetaValidationUtils.java <https://github.com/molgenis/molgenis/pull/3124#discussion_r31796751>:
> 
> > +	{
> > +		if (name.length() > MAX_ATTRIBUTE_LENGTH)
> > +		{
> > +			throw new MolgenisDataException("Attribute name [" + name + "] is too long: maximum length is "
> > +					+ MAX_ATTRIBUTE_LENGTH + " characters.");
> > +		}
> > +
> > +		if (!name.matches("[a-zA-Z0-9_#]+"))
> > +		{
> > +			throw new MolgenisDataException("Invalid characters in: [" + name
> > +					+ "] Only letters (a-z, A-Z), digits (0-9) and underscores (_) are allowed.");
> > +		}
> > +
> > +		if (Character.isDigit(name.charAt(0)))
> > +		{
> > +			throw new MolgenisDataException("Invalid name: [" + name + "] Names must start with a letter.");
> We have to allow starting with hash because of VCF (#CHROM). Underscore, I'm not sure.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/molgenis/molgenis/pull/3124/files#r31796751>.
>